io.github.andrewmkiss.Demeter.yml

```diff
@@ -0,0 +1,190 @@
+      - type: shell
+        commands:
+          # Have Flatpak run the GNU-compatible configure script.
+          - 'ln -s configure{.gnu,}'
+    # Restore write permission to the Perl libraries.
+    post-install:
+      - 'chmod -R u+w /app/lib/perl5'
+    # Clean up a bunch of stuff we don't need. Depending on your application,
+    # you may have to drop some of these (e.g. *.pod).
+    cleanup:
+      - '/bin/corelist'
+      - '/bin/cpan'
+      - '/bin/enc2xs'
+      - '/bin/encguess'
+      - '/bin/h2ph'
+      - '/bin/h2xs'
+      - '/bin/instmodsh'
+      - '/bin/json_pp'
+      - '/bin/libnetcfg'
+      - '/bin/perlbug'
+      - '/bin/perldoc'
+      - '/bin/perlivp'
+      - '/bin/perlthanks'
+      - '/bin/piconv'
+      - '/bin/pl2pm'
+      - '/bin/pod*'
+      - '/bin/prove'
+      - '/bin/ptar*'
+      - '/bin/shasum'
+      - '/bin/splain'
+      - '/bin/xsubpp'
+      - '/bin/zipdetails'
+      - '/include'
+      - '/man'
+      - '*.pod'
+
+  ## Using the dependencies from flatpak-cpan-generator
+  # Installs the modules generated by flatpak-cpan-generator.
+  # Note that *any* Makefile.PL-style modules MUST be installed in this one step,
+  # as once perllocal.pod is written in one module, it cannot be modified by others.
+  - name: perl-libs
+    disabled: false
+    buildsystem: simple
+    build-commands:
+      - 'perl-libs/install.sh'
+    # Same as with the Perl module, we need to restore write permission.
+    # However, -f is now passed to avoid errors from trying to touch files from the
+    # above module that are now marked as read-only.
+    post-install:
+      - 'chmod -Rf u+w /app/lib/perl5/site_perl'
+      - 'chmod u+w /app/bin/pdl*'
+    sources:
+      - perl-libs.yml
+    # This step should be customized based on the CPAN packages you're using.
+    cleanup:
+      - '/bin'
+      - '/man'
+
+  ## Demeter
+  - name: demeter
+    disabled: false
+    buildsystem: simple
+    build-options:
+      env:
+        - PERL5LIB=/run/build/demeter:/app/lib/perl5/site_perl
+        - PERLLIB=/app/lib/perl5/site_perl/5.36.1
+    build-commands:
+      - perl Build.PL
+      - sed -i "1653,+3s/25/35/" lib/Demeter/UI/Athena.pm
+      - ./Build
+      - ./Build install
+      - install -Dm 644 metainfo.xml /app/share/metainfo/io.github.andrewmkiss.Demeter.metainfo.xml
+      - install -Dm 644 /app/lib/perl5/site_perl/5.36.1/x86_64-linux/Demeter/share/Demeter_icon.png /app/share/icons/hicolor/48x48/apps/io.github.andrewmkiss.Demeter.png
+      - install -Dm 644 /app/lib/perl5/site_perl/5.36.1/x86_64-linux/Demeter/UI/Athena/share/athena_icon.png /app/share/icons/hicolor/48x48/apps/io.github.andrewmkiss.Demeter.Athena.png
+      - install -Dm 644 /app/lib/perl5/site_perl/5.36.1/x86_64-linux/Demeter/UI/Artemis/share/artemis_icon.png /app/share/icons/hicolor/48x48/apps/io.github.andrewmkiss.Demeter.Artemis.png
+      - ffmpeg -i Demeter.png -s 128x128 -frames:v 1 Demeter_128x128.png
```

Comment:
you don't need to do that. A 512x512 icon will be fine.

Reply:
`flatpak-builder-lint` wasn't happy until I got the right size image in the right location. I can remove this related code

Reply:
this is NOT the right location. This is the output of appstream-compose. if it fail it's because of something else. Like the lack of .desktop file.... (none of the one I see installed match the flatpak id)